src/bin/uhyve.rs

```diff
@@ -490,17 +513,83 @@ fn run_uhyve() -> i32 {
 
 	let mut app = Args::command();
 	let mut args = Args::parse();
+	let stats;
+
+	let res = match args.guest.guest.clone() {
+		Guest::Kernel { kernel_path } => {
+			load_vm_config(&mut args);
 
-	load_vm_config(&mut args);
+			stats = args.uhyve.stats.unwrap_or_default();
+			let affinity = args.cpu.clone().get_affinity(&mut app);
+			let params = Params::from(args);
 
-	let stats = args.uhyve.stats.unwrap_or_default();
-	let kernel_path = args.guest.kernel.clone();
-	let affinity = args.cpu.clone().get_affinity(&mut app);
-	let params = Params::from(args);
+			let vm = UhyveVm::new(kernel_path, params).unwrap_or_else(|e| panic!("Error: {e}"));
+
+			vm.run(affinity)
+		}
+		Guest::Image { image_path } => {
+			// Unfortunately, we have to read the image twice (unless we figure out a way to cache it...)
+			let data =
```

Comment:
This functionality is probably best located in uhyvelib as a separate function akin to `Vm::new`. This would probably also reduce the complexity and number of duplications.